app/src/main/java/com/example/atheneum/fragments/EditProfileFragment.java

```diff
@@ -114,6 +114,9 @@ public void onClick(final View v) {
                 final User user = userViewModel.getUserLiveData().getValue();
                 // Update the user based on form input
                 user.setPhoneNumber(phoneNumberField.getText().toString());
+                if (emailAddressField.getVisibility() != View.GONE) {

```

Comment:
Have to update the username but can only do it when the emailAddressField is there since we set it to GONE during registration. If this check is not there, you could end up with an empty username.

Reply:
So where is EditProfileFragment used?

Reply:
The EditProfileFragment is used in both CompleteRegistrationActivity (which is triggered only once right after the user signs-up for the first time) and in the EditProfileActivity which is triggered when a user is trying to edit their profile any other time. The EditProfileFragment contains the form for editing a profile. All those two activites do is inflate the EditProfileFragment and set the onEditProfileSuccess and onEditProfileFailure listeners. It was made as a fragment instead of an activity since the two activities basically share that form but then have different behaviour when the EditProfile form is saved.